File: src/wsTripXML/Code/OTAClasses/OTACruiseCreateBooking.vb
Option Strict Off
Option Explicit On 

Imports System.Xml.Serialization
Imports wsTripXML.wsTravelTalk.wmCruiseBooking

Namespace wsTravelTalk.wmCruiseCreateBooking

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class SailingInfo

        '<remarks/>
        Public SelectedSailing As SelectedSailing

        '<remarks/>
        Public InclusivePackageOption As InclusivePackageOption

        '<remarks/>
        Public Currency As Currency

        '<remarks/>
        Public SelectedCategory As SelectedCategory

        '<remarks/>
        Public DeparturePort As DeparturePort

        '<remarks/>
        Public ArrivalPort As ArrivalPort

        '<remarks/>
        Public FaxNotification As FaxNotification
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class FaxNotification

        '<remarks/>
        <System.Xml.Serialization.XmlElementAttribute("NotificationType")> _
        Public NotificationType() As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public PhoneLocationType As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public PhoneTechType As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public CountryAccessCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public AreaCityCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public PhoneNumber As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public Extension As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public PIN As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class DeparturePort

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public LocationCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlTextAttribute()> _
        Public Value As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class ArrivalPort

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public LocationCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlTextAttribute()> _
        Public Value As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class ReservationID

        '<remarks/>
        Public CompanyName As CompanyName

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public URL As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public Type As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public Instance As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public ID As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public ID_Context As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public SyncDateTime As String
    End Class

End Namespace

File: src/wsTripXML/Code/OTAClasses/OTACruiseCreateBooking.cs

using System.Xml.Serialization;
using wsTripXML.wsTravelTalk.wmCruiseBooking;

namespace wsTripXML.wsTravelTalk.wmCruiseCreateBooking
{

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class SailingInfo
    {

        // <remarks/>
        public SelectedSailing SelectedSailing;

        // <remarks/>
        public InclusivePackageOption InclusivePackageOption;

        // <remarks/>
        public Currency Currency;

        // <remarks/>
        public SelectedCategory SelectedCategory;

        // <remarks/>
        public DeparturePort DeparturePort;

        // <remarks/>
        public ArrivalPort ArrivalPort;

        // <remarks/>
        public FaxNotification FaxNotification;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class FaxNotification
    {

        // <remarks/>
        [XmlElement("NotificationType")]
        public string[] NotificationType;

        // <remarks/>
        [XmlAttribute()]
        public string PhoneLocationType;

        // <remarks/>
        [XmlAttribute()]
        public string PhoneTechType;

        // <remarks/>
        [XmlAttribute()]
        public string CountryAccessCode;

        // <remarks/>
        [XmlAttribute()]
        public string AreaCityCode;

        // <remarks/>
        [XmlAttribute()]
        public string PhoneNumber;

        // <remarks/>
        [XmlAttribute()]
        public string Extension;

        // <remarks/>
        [XmlAttribute()]
        public string PIN;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class DeparturePort
    {

        // <remarks/>
        [XmlAttribute()]
        public string LocationCode;

        // <remarks/>
        [XmlText()]
        public string Value;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class ArrivalPort
    {

        // <remarks/>
        [XmlAttribute()]
        public string LocationCode;

        // <remarks/>
        [XmlText()]
        public string Value;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class ReservationID
    {

        // <remarks/>
        public Code.CompanyName CompanyName;

        // <remarks/>
        [XmlAttribute()]
        public string URL;

        // <remarks/>
        [XmlAttribute()]
        public string Type;

        // <remarks/>
        [XmlAttribute()]
        public string Instance;

        // <remarks/>
        [XmlAttribute()]
        public string ID;

        // <remarks/>
        [XmlAttribute()]
        public string ID_Context;

        // <remarks/>
        [XmlAttribute()]
        public string SyncDateTime;
    }

}